File: test_data/code/python/1422-maximum-score-after-splitting-a-string.py
Python
class Solution:
    def maxScore(self, s: str) -> int:
        n = len(s)
        total_zeros = sum(1 for c in s if c == '0')
        max_score = 0
        left_zeros = 0
        for right_zero in range(n):
            left_zeros += 1 if s[right_zero] == '0' else 0
            max_score = max(max_score, left_zeros + total_zeros - left_zeros)
        return max_score

File: test_data/code/java/1422-maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
        int max = 0;
        for (int i = 1; i < s.length() - 1; i++) {
            int leftOnes = 0, rightZeros = 0;
            for (int j = 0; j < i; j++) {
                if (s.charAt(j) == '1') {
                    leftOnes++;
                }
            }
            for (int k = i; k < s.length(); k++) {
                if (s.charAt(k) == '0') {
                    rightZeros++;
                }
            }
            max = Math.max(max, leftOnes + rightZeros);
        }
        return max;
    }
}